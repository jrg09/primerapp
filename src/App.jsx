import React from 'react';
import ReactDOM from 'react-dom/client';
import PropTypes from 'prop-types';
import { Header } from './components/Header';
import { Content } from './components/Content';
import { Footer } from './components/Footer.jsx';
import './styles.css'


const App = ({title,subtitle,name, initial}) => {
    return(
        <>
            <Header title={title} veces={3}/>
            <Content subtitle={subtitle} initial={initial}/>
            <Footer/>
        </>
       /*
        <>
            <h2 data-testid="test-title">{title}</h2>
            <p>{subtitle}</p>
            <p>{name}</p>
            <hr/>
            <span>jorge</span>
            <p>{title}</p>
        </>
        */
    );
}

App.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

export default App;
