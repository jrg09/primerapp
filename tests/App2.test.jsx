import { render,screen } from "@testing-library/react";
import App from "../src/App";

describe('Pruebas en <App/>', () => {
    
    const title = "Hola, Soy Jorge";

    test('debe hacer match con el snapshot', () => {

        const {container} = render(<App title={title}/>);

        expect(container).toMatchSnapshot();
    }); 

    test('should mostrar mensaje "hola sour x"', async () => { 
        render(<App title={title}></App>);
        //expect(screen.getByText(title)).toBeTruthy();
        const actual = (await screen.findAllByText(title)).length;
        console.log('actual',actual);
        expect(actual).toBe(1);
     })
}); 