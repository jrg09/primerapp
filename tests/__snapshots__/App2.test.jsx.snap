// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App/> debe hacer match con el snapshot 1`] = `
<div>
  <header>
    <h1>
      Hola, Soy Jorge
    </h1>
    <div>
      se repite: 
      <b>
        3
      </b>
      :
      <span>
        0
        span 
      </span>
      <span>
        1
        span 
      </span>
      <span>
        2
        span 
      </span>
    </div>
  </header>
  <div
    id="content"
  >
    <h1>
      Content Section
    </h1>
    <p>
      <code>
        {"title":"jjj üèé","msg":"jorge hola"}
      </code>
    </p>
    <p>
      <b>
        jorge 
        jorge hola
      </b>
    </p>
    <p>
      subtitulo: 
    </p>
    <h2>
      Counter
    </h2>
    <h2>
      Inicial: 
      0
    </h2>
    <h3>
      <span
        data-testid="span-test"
      >
        0
      </span>
    </h3>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      reset
    </button>
  </div>
  <footer>
    footer
  </footer>
</div>
`;
